server.js

```diff
@@ -58,19 +62,60 @@ MongoClient.connect('mongodb://localhost:4201/tutorial-builder', (err, database)
 SERVER Routes
 
 */
+app.put('/changepass', (req, res) => {
+	db.collection('users').findOne({username: req.body.username}).then(user => {
+		if (!user) {
+			return res.send(boom.unauthorized('invalid username'));
+		}
+
+		return bcrypt.compare(req.body.password, user.password, function(err, resp) {
+			if(resp) {
```

Comment:
When the passwords are not the same, what does this respond with? does it error out or is the response different?

I would instead do something like: 
```
if (err) {
  console.log(err);
  res.send(boom.badImplementation(err)) //or check what the error is
}

bcrypt.hash ... //I think safe to assume there's a response if err is null 
```